### PR TITLE
created new hashing data structure

### PR DESCRIPTION
**Pull request by furahadamien**
created a new data structure that uses open addressing for hashing. uses a quadratic hashing technique to generate the hash key

### CODE REVIEW

File: DataStructures/HashMap/Hashing/Open_Addressing.java [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    public int tableSize; // number of SLOTS AVAILABLE
+    public int randomNumber; // the default random number
+    int index;
+    int hasher;
+    public int[] Table;
+
+    protected Open_Addressing(int index, int seed) {
+        this.index = index;
+        this.hasher = (int) (index - 1) / 2 + 1;
+        this.tableSize = power2(hasher);
```
Comment (yanglbme) [inline DataStructures/HashMap/Hashing/Open_Addressing.java]: where's the method `power2()` ?

File: DataStructures/HashMap/Hashing/Open_Addressing.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
+/**
+ * Here I imnplement a hashing function
+ * the Open_Addressing object uses an array for hashing
+ * using quadratic hashing, we make it possible to add an element and retrieve its value from the array in constant time
+ */
+public class Open_Addressing {
```
Comment (yanglbme) [inline DataStructures/HashMap/Hashing/Open_Addressing.java]: It should be `OpenAddressing`, not `Open_Addressing`.

File: DataStructures/HashMap/Hashing/OpenAddressing.java [hunk 3]
```diff
@@ -102,4 +102,11 @@ public boolean isDuplicate(int [] table, int key){
         }
         return found;
     }
+
+     /**
+     * Calculate 2^index
+     */
+    public static int power2(int index) {
+        return (int) Math.pow(2, index);
+    }
```
Comment (furahadamien) [inline DataStructures/HashMap/Hashing/OpenAddressing.java]: added the power function that was requested

File: DataStructures/HashMap/Hashing/OpenAddressing.java [hunk 4]
```diff
@@ -3,15 +3,15 @@
  * the Open_Addressing object uses an array for hashing
  * using quadratic hashing, we make it possible to add an element and retrieve its value from the array in constant time
  */
-public class Open_Addressing {
+public class OpenAddressing {
```
Comment (furahadamien) [inline DataStructures/HashMap/Hashing/OpenAddressing.java]: renamed the class

File: DataStructures/HashMap/Hashing/Open_Addressing.java (hunk 2 above)
Comment (furahadamien) [reply DataStructures/HashMap/Hashing/Open_Addressing.java]: Changed the name in my second commit

File: DataStructures/HashMap/Hashing/Open_Addressing.java (hunk 1 above)
Comment (furahadamien) [reply DataStructures/HashMap/Hashing/Open_Addressing.java]: I  had forgotten the function in my first commit. Now added

**Pull request closed without merging**